pandas/tests/indexes/test_multi.py

```diff
@@ -1717,6 +1725,12 @@ def test_from_tuples(self):
                                'from empty list',
                                MultiIndex.from_tuples, [])
 
+        result = MultiIndex.from_tuples([], names=['a', 'b'])
```

Comment:
I agree with this. It should raise because you have a mismatch between the names and what you passed. If you had NOT passed an empty list it would be ok.